### PR TITLE
chore: run test container if jdbc repository has been modified

### PR DESCRIPTION
**Pull request by ytvnr**
Add a should_build.sh to be able to check if a folder has git diff and know if we should buiild/test it

## What it does 

1. Check if we are currently building master (always execute)
2. Check if we are in a PR and get the target branch
3. Do a `git diff `between your PR and the target branch
4. Check if the folder you want to test/build has diff
5. If diffs, continue the step

## Usage

Let's say you have a `backend` folder and a `front` folder in your repo. You don't want to build the `front` if you only provides modifications in the `backend`

In your step configuration, simply use the script like that:
```
- run:
          name: Check if front should be built
          command: |
            ./.circleci/should_build.sh front || circleci step halt
```

If there is no diff in `front` folder, then the circleci step will stop here (`halt` command: https://circleci.com/docs/2.0/configuration-reference/#ending-a-job-from-within-a-step)

~~⚠️ with this version, the diff is only applied between the pull request's branch and master. So for support version (3.5.x for example) the diff will be done with master~~
✅ The git diff is done against the target branch of your PR

Here is an example with dummy steps:
- > no modification inside repository-jdbc: https://app.circleci.com/pipelines/github/gravitee-io/gravitee-api-management/873/workflows/d4559ee4-2105-4a53-94d2-68144452b3a3/jobs/2101
- > modifying reposiroty-jdbc: https://app.circleci.com/pipelines/github/gravitee-io/gravitee-api-management/874/workflows/42d1cb20-b292-43f7-870a-cf97ecd2b9d5/jobs/2105

### CODE REVIEW

File: .circleci/should_build.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        url="https://api.github.com/repos/gravitee-io/${CIRCLE_PROJECT_REPONAME}/pulls/$pr_number"
+        target_branch=$(curl "$url" | ./jq '.base.ref' | tr -d '"')
+    fi
+    echo "Target branch is $target_branch"
+    # Check the diff between target branch and current branch
+    git diff --name-only origin/${target_branch}...${CIRCLE_BRANCH} | grep "^$1" > /dev/null 2>&1
+}
+
+# Build systematically master branch
+if [[ "${CIRCLE_BRANCH}" = "master" ]]; then exit 0; fi
```
Comment (gaetanmaisse) [inline .circleci/should_build.sh:18]: we should do the same for the support branches too no?

File: .circleci/config.yml [hunk 2]
```diff
@@ -94,6 +94,39 @@ jobs:
       - store_artifacts:
           path: ~/test-results/junit
 
+  test-repository-jdbc:
+    machine:
+      image: ubuntu-2004:202107-02
+    resource_class: xlarge
```
Comment (gaetanmaisse) [inline .circleci/config.yml:100]: `xlarge`, toujours plus 😛

File: .circleci/config.yml [hunk 3]
```diff
@@ -94,6 +94,39 @@ jobs:
...
+    steps:
+      - checkout
+      - run:
+          name: Check if gravitee-apim-repository-jdbc should be built
```
Comment (gaetanmaisse) [inline .circleci/config.yml:104]: `Check if gravitee-apim-repository-jdbc should be built` it's maybe more accurate to rename this and the script to `Check if xxxxx has changes`

**Review by gaetanmaisse — COMMENTED** [review]
It looks great! 💪🏻

File: .circleci/should_build.sh (hunk 1 above)
Comment (ytvnr) [reply .circleci/should_build.sh:18]: yes, it would be preferable
a regex maybe ?

File: .circleci/config.yml (hunk 2 above)
Comment (ytvnr) [reply .circleci/config.yml:100]: copy paste :p

File: .circleci/config.yml (hunk 3 above)
Comment (ytvnr) [reply .circleci/config.yml:104]: yes, it's just a draft as an example :)

**Comment by gaetanmaisse** [comment]
Replaced by https://github.com/gravitee-io/gravitee-api-management/pull/862

**Pull request closed without merging**